src/api/gocdevents/index.test.ts

```diff
@@ -20,11 +20,36 @@ describe('gocdevents', function () {
     gocdevents.on('stage', stageSpy);
     gocdevents.on('agent', agentSpy);
 
-    gocdevents.emit('stage', {});
-    gocdevents.emit('agent', {});
+    gocdevents.emit('stage', {
+      data: {
+        pipeline: { name: 'test-name', group: 'test-group' },
+      },
+    });
+    gocdevents.emit('agent', {
+      data: {
+        pipeline: { name: 'test-name', group: 'test-group' },
+      },
+    });
 
     expect(starSpy).toHaveBeenCalledTimes(2);
     expect(stageSpy).toHaveBeenCalledTimes(1);
     expect(agentSpy).toHaveBeenCalledTimes(1);
   });
+
+  it('do not emit stage when no group is provided', () => {
+    const starSpy = jest.fn();
+    const stageSpy = jest.fn();
+    const agentSpy = jest.fn();
+
+    gocdevents.on('*', starSpy);
+    gocdevents.on('stage', stageSpy);
+    gocdevents.on('agent', agentSpy);
+
+    gocdevents.emit('stage', { data: { pipeline: { name: 'test-name' } } });
```

Comment:
Can you add a test case where you add the group and ensure the star and stage get it